doc/design/upgrade.md

```diff
@@ -38,14 +59,16 @@ If anything wrong happens during upgrade, vic-machine will rollback VCH to origi
 
 ### Downgrade
 
-For any running VCH, VIC admin can downgrade existing VCH to an old version. (I assume this function is only used for upgrade roll back, we might not want to recommend downgrade for anything working well, then the downgrade scope will be same to upgrade versions)
-
-Verified docker-engine upgrade/downgrade, which works well between 1.11.2 and 1.12. The running container will be stopped after upgrade/downgrade, but is good to start again in new version. And all images and containers information is not lost. So from technical point of view, we should support same story.
+Verified docker-engine upgrade/downgrade, which works well between 1.11.2 and 1.12. The running container will be stopped after upgrade/downgrade, but is good to start again in new version. And all images and containers information is not lost. But think of the complexity to support version downgrade, we'll not go with this at this moment, unless there is explicity user requirement.
```

Comment:
>If containers are created with the upgraded VCH, what happens to them after rollback? 

I guess here you mean if container configuration upgrade failed, and then we rollback to old version, right? Cause if VCH upgrade is rolled back, then nothing will happen for container.
Currently, we only support guestinfo upgrade for container VM, not related to key/value store update. 
So actually two things related
- there is container related data persisted in key/value store, which is not backward compatible, so we might have function loss for old version's container. I'll figure out if we have that right now.
- container guestinfo data migration failed, and rolled back. Which is bad for our case, cause we don't want to roll back VCH upgrade in this case, that means we lost all support for that container. Portlayer need to handle that failure, but eventually, from docker client, user might not be able to see that container anymore.